frontend/src/core/hooks/useStatus.js

```diff
@@ -0,0 +1,89 @@
+import { useMutation } from "react-query";
+import { useToast } from "./";
+import { AuthService } from "../../core/services";
+
+const useStatus = () => {
+	const toast = useToast();
+
+	const { mutate: apiServerStatus, data: apiServerData } = useMutation(
```

Comment:
I don't understand why you are using `useMutation` instead of `useQuery` here.

Suggest changing this to `useQuery` with some caching so that status information is cached (with some low TTL) and is immediately available to various components.

See this GitHub issue for more details:
https://github.com/tannerlinsley/react-query/discussions/580#discussioncomment-26019

Reply:
Also, a helpful guide to `useQuery` vs. `useMutation` from the author of `react-query`: https://twitter.com/tannerlinsley/status/1324384797939003393